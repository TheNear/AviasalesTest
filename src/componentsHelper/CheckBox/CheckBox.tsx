import React from "react";
import style from "./CheckBox.module.css";

interface ICheckBox {
  text: string;
  name: string;
}

const CheckBox: React.FC<ICheckBox> = ({ text, name }) => {
  return (
    <li>
      <label className={style.label} htmlFor={name}>
        <input type="checkbox" className={style.input} id={name} name={name} value={name} />
        <span className={style.checkbox} />
        {text}
      </label>
    </li>
  );
};

export default CheckBox;
