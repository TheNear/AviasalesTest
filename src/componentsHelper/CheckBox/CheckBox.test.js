import React from "react";
import { shallow } from "enzyme";
import CheckBox from "./CheckBox";

describe("Common Checkbox render", () => {
  it("Render empty Checkbox", () => {
    const wrapper = shallow(<CheckBox />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Render Checkbox with props", () => {
    const wrapper = shallow(<CheckBox text="Check Box Title" name="check-box-name" />);
    expect(wrapper).toMatchSnapshot();
  });
});
