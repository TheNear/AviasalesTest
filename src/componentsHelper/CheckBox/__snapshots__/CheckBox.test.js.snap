// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Checkbox render Render Checkbox with props 1`] = `
<li>
  <label
    className="label"
    htmlFor="check-box-name"
  >
    <input
      className="input"
      id="check-box-name"
      name="check-box-name"
      type="checkbox"
      value="check-box-name"
    />
    <span
      className="checkbox"
    />
    Check Box Title
  </label>
</li>
`;

exports[`Common Checkbox render Render empty Checkbox 1`] = `
<li>
  <label
    className="label"
  >
    <input
      className="input"
      type="checkbox"
    />
    <span
      className="checkbox"
    />
  </label>
</li>
`;
